Portal/Controllers/WasteController.cs

```diff
@@ -27,12 +27,12 @@ public async Task<IActionResult> DuringWhichMonth(int? id)
         }
 
         [HttpGet]
-        public IActionResult WhatHaveYouDoneWaste(int? id)
+        public async Task<IActionResult> WhatHaveYouDoneWaste(int? id)
         {
             if (id == null)
                 throw new ArgumentNullException("id");
```

Comment:
this should be a NotFound not an exception. Proving no ID will represent an incorrect URL, not exception to be thrown